presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchConfig.java

```diff
@@ -196,6 +199,48 @@ public ElasticsearchConfig setNodeRefreshInterval(Duration nodeRefreshInterval)
         return this;
     }
 
+    @NotNull
+    public Duration getRequestQueuedTimeout()
+    {
+        return requestQueuedTimeout;
+    }
+
+    @Config("elasticsearch.request-queued-timeout")
+    @ConfigDescription("The maximum time a request can wait for a connection to become available")
+    public ElasticsearchConfig setRequestQueuedTimeout(Duration requestQueuedTimeout)
+    {
+        this.requestQueuedTimeout = requestQueuedTimeout;
+        return this;
+    }
```

Comment:
I had it hard-coded originally and right before pushing I thought that this setting might provide a back-pressure mechanism of sorts (if you were to set it to a reasonable value, it would fail queries quickly, should ES become overloaded and take a long time to respond).  But I don't feel strongly about it - I'll drop it if you think it's too much.